### PR TITLE
Compute BitConverter.IsLittleEndian correctly under mono.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/Common/src/CoreLib/System/BitConverter.cs [hunk 1]
```diff
@@ -19,10 +19,23 @@ public static class BitConverter
         // This field indicates the "endianess" of the architecture.
         // The value is set to true if the architecture is
         // little endian; false if it is big endian.
+#if MONO
+		[Intrinsic]
+		public static readonly bool IsLittleEndian;
+
+		static BitConverter () {
+			unsafe {
+				ushort i = 0x1234;
```
Comment (jaykrell) [inline src/Common/src/CoreLib/System/BitConverter.cs]: I think `int i = 1` would  be clearest here.

Comment (jaykrell) [reply src/Common/src/CoreLib/System/BitConverter.cs]: And then just check the low  byte  for `0` or `1`. `1234` is unnecessarily cute.

Comment (jaykrell) [reply src/Common/src/CoreLib/System/BitConverter.cs]: What does the instrinic attribute mean?

Comment (vargaz) [reply src/Common/src/CoreLib/System/BitConverter.cs]: It just a market which means the JIT is going to optimize it in some way.

**Comment by jaykrell** [comment]
I don’t think this should be mono-specific & I think the integer should be ‘int i = 1’ and then check low byte for 0 or 1, but ok.

File: src/Common/src/CoreLib/System/BitConverter.cs (hunk 1 above)
Comment (marek-safar) [reply src/Common/src/CoreLib/System/BitConverter.cs]: Are you sure the explicit static ctor won't break AOT inlining for IsLittleEndian ?

File: src/Common/src/CoreLib/System/BitConverter.cs [hunk 2]
```diff
@@ -19,10 +19,23 @@ public static class BitConverter
...
+				byte *b = (byte*)&i;
+				IsLittleEndian = (*b == 0x34);
+			}
+		}
+#else
```
Comment (marek-safar) [inline src/Common/src/CoreLib/System/BitConverter.cs]: It'd be better not to touch CoreFX code and just if-def this out completely for #if MONO and make the type partial with Mono implementation in mcs/class/System/BitConverter.cs

**Pull request merged**